src/cpp/xmlparser/XMLProfileManager.cpp

```diff
@@ -81,6 +68,33 @@ XMLP_ret XMLProfileManager::fillSubscriberAttributes(const std::string &profile_
     return XMLP_ret::XML_OK;
 }
 
+XMLP_ret XMLProfileManager::fillTopicAttributes(const std::string& profile_name, TopicAttributes& atts)
+{
+    topic_map_iterator_t it = m_topic_profiles.find(profile_name);
+    if (it == m_topic_profiles.end())
+    {
+        logError(XMLPARSER, "Profile '" << profile_name << "' not found");
+        return XMLP_ret::XML_ERROR;
+    }
+    atts = *(it->second);
+    return XMLP_ret::XML_OK;
+}
+
+void XMLProfileManager::getDefaultParticipantAttributes(ParticipantAttributes& participant_attributes)
+{
+    participant_attributes = default_participant_attributes;
+}
+
+void XMLProfileManager::getDefaultPublisherAttributes(PublisherAttributes& publisher_attributes)
+{
+    publisher_attributes = default_publisher_attributes;
+}
+
+void XMLProfileManager::getDefaultSubscriberAttributes(SubscriberAttributes& subscriber_attributes)
+{
+    subscriber_attributes = default_subscriber_attributes;
+}
+
```

Comment:
I think we should also add a getDefaultTopicAttributes method

Reply:
Done!